homeassistant/components/konnected/config_flow.py

```diff
@@ -0,0 +1,561 @@
+DATA_SCHEMA_SWITCH_OPTIONS[vol.Optional("pause")] = int
+DATA_SCHEMA_SWITCH_OPTIONS[vol.Optional("repeat")] = int
+
+DATA_SCHEMA_OPTIONS = {
+    "Binary Sensor": DATA_SCHEMA_BIN_SENSOR_OPTIONS,
+    "Sensor": DATA_SCHEMA_SENSOR_OPTIONS,
+    "Switch": DATA_SCHEMA_SWITCH_OPTIONS,
+}
+
+BINARY_SENSOR_SCHEMA = vol.All(
+    vol.Schema(
+        {
+            vol.Exclusive(CONF_ZONE, "s_zone"): vol.In(ZONES),
+            vol.Required(CONF_ZONE): vol.In(ZONES),
+            vol.Required(CONF_TYPE): DEVICE_CLASSES_SCHEMA,
+            vol.Optional(CONF_NAME): cv.string,
+            vol.Optional(CONF_INVERSE, default=False): cv.boolean,
+        }
+    )
+)
+
+SENSOR_SCHEMA = vol.All(
+    vol.Schema(
+        {
+            vol.Exclusive(CONF_ZONE, "s_zone"): vol.In(ZONES),
+            vol.Required(CONF_ZONE): vol.In(ZONES),
+            vol.Required(CONF_TYPE): vol.All(vol.Lower, vol.In(["dht", "ds18b20"])),
+            vol.Optional(CONF_NAME): cv.string,
+            vol.Optional(CONF_POLL_INTERVAL, default=3): vol.All(
+                vol.Coerce(int), vol.Range(min=1)
+            ),
+        }
+    )
+)
+
+SWITCH_SCHEMA = vol.All(
+    vol.Schema(
+        {
+            vol.Exclusive(CONF_ZONE, "s_zone"): vol.In(ZONES),
+            vol.Required(CONF_ZONE): vol.In(ZONES),
+            vol.Optional(CONF_NAME): cv.string,
+            vol.Optional(CONF_ACTIVATION, default=STATE_HIGH): vol.All(
+                vol.Lower, vol.Any(STATE_HIGH, STATE_LOW)
+            ),
+            vol.Optional(CONF_MOMENTARY): vol.All(vol.Coerce(int), vol.Range(min=10)),
+            vol.Optional(CONF_PAUSE): vol.All(vol.Coerce(int), vol.Range(min=10)),
+            vol.Optional(CONF_REPEAT): vol.All(vol.Coerce(int), vol.Range(min=-1)),
+        }
+    )
+)
+
+DEVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ID): cv.matches_regex("[0-9a-f]{12}"),
+        vol.Optional(CONF_BINARY_SENSORS): vol.All(
+            cv.ensure_list, [BINARY_SENSOR_SCHEMA]
+        ),
+        vol.Optional(CONF_SENSORS): vol.All(cv.ensure_list, [SENSOR_SCHEMA]),
+        vol.Optional(CONF_SWITCHES): vol.All(cv.ensure_list, [SWITCH_SCHEMA]),
+        vol.Optional(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT): cv.port,
+        vol.Optional(CONF_BLINK, default=True): cv.boolean,
+        vol.Optional(CONF_DISCOVERY, default=True): cv.boolean,
+    }
+)
+
+
+@callback
+def configured_hosts(hass):
+    """Return a set of the configured hosts."""
+    return set(
+        entry.data.get("host") for entry in hass.config_entries.async_entries(DOMAIN)
+    )
+
+
+@callback
+def configured_devices(hass):
+    """Return a set of the configured devices."""
+    return set(entry.data["id"] for entry in hass.config_entries.async_entries(DOMAIN))
+
+
+class KonnectedFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for NEW_NAME."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
```

Comment:
what does this indicate?
the device is really a "local push"

Reply:
Nice catch - I'll change this.